### ISSUE TITLE
[no-base-to-string] complains about intersection types

### ISSUE THREAD

**Issue by bogdanb** [typescript-eslint/typescript-eslint#2162]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-base-to-string": <default>
  }
}
```

```TS
export interface Brand {}
export function test(v: string & Brand): string {
  return `${v}`; // v will evaluate to '[object Object]' when stringified
}
```

**Expected Result**

The error shown in the comment is reported.

**Actual Result**

The error should not be reported, because `v` is known to be a `string`.

**Additional Info**

This is a very simplified case just to allow quick reproduction. In practice I encountered this using [branded types from the io-ts library](https://github.com/gcanti/io-ts/blob/master/index.md#branded-types--refinements). Note that their technique can be used with other types, not just strings.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.31.0` |
| `@typescript-eslint/parser`        | `2.31.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.14.2` |
